scipy/signal/signaltools.py

```diff
@@ -1662,6 +1662,16 @@ def cmplx_sort(p):
     indx : ndarray
         Array of indices needed to sort the input `p`.
 
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> vals = [1, 4, 1+1.j, 3]
+    >>> p_sorted, indx = signal.cmplx_sort(vals)
+    >>> p_sorted
+    array([1.0+0.j, 1.0+1.j, 3.+0.j, 4.+0.j])
```

Comment:
Is this the actual output that you get when you display `p_sorted`?   With numpy 1.14.1, I get

    array([1.+0.j, 1.+1.j, 3.+0.j, 4.+0.j])

Note that the real parts of the first two values are displayed as `1.`, not `1.0`.

With numpy 1.13.1, I get

    array([ 1.+0.j,  1.+1.j,  3.+0.j,  4.+0.j])

Reply:
I just checked it and indeed that's not the case.
The output with version 1.8.2 is: 
`array([ 1.+0.j,  1.+1.j,  3.+0.j,  4.+0.j])`
Just as you pointed out. I just pushed desirable changes into this branch.